src/components/terminal/input/terminal-input.component.js

```diff
@@ -33,27 +38,62 @@ export const TerminalInput = props => {
         }
     }
 
+    const onKeyDown = event$ => {
+        const element = event$.target;
+
+        // Preventing deletion of the prompt by backspace or delete keys
+        // keyCode 8: Backspace
+        // keyCode 46: Delete
+        if (element.selectionStart <= promptText.length ){
+            if (event$.keyCode === 8 || event$.keyCode === 46){
+                if (element.selectionStart === element.selectionEnd)
+                    event$.preventDefault()
+                console.log("ACTION FORBIDDEN: Trying to modify prompt")
+            }
+            positionCursor(element)
+            return
+        }
+        if (event$.keyCode === 13);
+            // TODO process command
+
+    }
+
+    onClick = e => { if (e.target.selectionStart <= promptText.length) positionCursor(e.target) }
+
+    onFocus = e => positionCursor(e.target)
+
+    const positionCursor = element => element.selectionStart = promptText.length
+
     const [ inputText, setInputText ] = useState('');
+    const [ promptText, setPromptText ] = useState('( base ) Alfred-MBP:~ alfredenewman$ ');
 
     
     useEffect( () => {
+        const element = document.getElementById("promptInput")
+        
+        element.value = promptText
+
+        positionCursor(element)
+
         window.addEventListener('keyup', onKeyUp );
+        document.getElementById('promptInput').addEventListener('keydown', onKeyDown );
+        document.getElementById('promptInput').addEventListener('click', onClick );
 
-        return () =>
-          window.removeEventListener('keyup', onKeyUp );
+        return () => {
+            window.removeEventListener('keyup', onKeyUp );
+            document.getElementById('promptInput').removeEventListener('keydown', onKeyDown );
+            document.getElementById('promptInput').removeEventListener('click', onClick );
+        }
       }
```

Comment:
consider passing dependency values into an array for change detection for performance.
try inputText.